tools/otci/otci/otci.py

```diff
@@ -2550,9 +2550,29 @@ def diag_cw_stop(self):
         """Stop transmitting continuous carrier wave."""
         self.execute_command('diag cw stop')
 
-    def diag_frame(self, frame: str):
+    def diag_frame(self,
+                   frame: str,
+                   max_csma_backoffs: Optional[int] = None,
+                   csma_ca_enabled: bool = False,
+                   rx_channel_after_tx_done: Optional[int] = None,
+                   tx_delay: Optional[int] = None,
+                   tx_power: Optional[int] = None,
+                   max_frame_retries: Optional[int] = None,
+                   is_security_processed: bool = False,
+                   is_header_updated: bool = False):
         """Set the frame (hex encoded) to be used by `diag send` and `diag repeat`."""
-        self.execute_command(f'diag frame {frame}')
+        command = f'diag frame '
+        command += f'-b {max_csma_backoffs} ' if max_csma_backoffs is not None else ''
```

Comment:
nit: There seems to be a common pattern 'if the argument is not `None` then append a command line option'. It seems more readable to make a helper function.